src/config/field-overrides/dataSet.js

```diff
@@ -52,39 +54,149 @@ class RenderAsTabsSettings extends React.Component {
-                    style={{margin: '10px'}}
+            <div>
+                <Checkbox
+                    labelText={this.translate('render_as_tabs')}
+                    value={props.value}
+                    onChange={this.onRenderAsTabsChanged}
                 />
-            </RadioButtonGroup>}
+                {props.value &&
+                    <RadioButtonGroup
+                        onChange={this.onDisplayOptionsChanged}
+                        name="tabsDirection"
+                        defaultSelected={
+                            (state.displayOptions && state.displayOptions.tabsDirection) || 'horizontal'}
+                    >
+                        <RadioButton
+                            key='horizontal'
+                            value='horizontal'
+                            label={this.translate('horizontal')}
+                            style={{margin: '10px'}}
+                        />
+                        <RadioButton
+                            key='vertical'
+                            value='vertical'
+                            label={this.translate('vertical')}
+                            style={{margin: '10px'}}
+                        />
+                    </RadioButtonGroup>}
+            </div>
+            <div>
+            <Checkbox
+                labelText={this.translate('add_custom_text')}
+                value={state.displayOptions && state.displayOptions.customText !== undefined}
+                onChange={this.onAddCustomTextChanged}
+            />
+
+            {state.displayOptions && state.displayOptions.customText &&
+                <RadioButtonGroup
+                    onChange={this.onCustomTextAlignmentChanged}
+                    name="customTextAlignment"
+                    defaultSelected={
+                        (state.displayOptions && state.displayOptions.customText && state.displayOptions.customText.align) || 'left'}
+                >
+                    <RadioButton
+                        key='left'
+                        value='left'
+                        label={this.translate('left')}
+                        style={{margin: '10px'}}
+                    />
+                    <RadioButton
+                        key='center'
+                        value='center'
+                        label={this.translate('center')}
+                        style={{margin: '10px'}}
+                    />
+                    <RadioButton
+                        key='right'
+                        value='right'
+                        label={this.translate('right')}
+                        style={{margin: '10px'}}
+                    />
+                </RadioButtonGroup>}
+
+                {state.displayOptions && state.displayOptions.customText &&
+                <div style={customTextCssStyles}>
+                    <TextField
+                    value={(this.state.displayOptions && state.displayOptions.customText &&
```

Comment:
is there a reason to use `this.state` vs `state` (applies a couple of other places within the render too). It seems like `const state = this.state` is declared above

Reply:
ehehe tiredness ;) changed